app/controllers/api/v1/admin/server_rooms_controller.rb

```diff
@@ -4,18 +4,24 @@ module Api
   module V1
     module Admin
       class ServerRoomsController < ApiController
+        # GET /api/v1/admin/server_rooms.json
         def index
-          rooms = Room.all.search(params[:search])
+          rooms = Room.includes(:user).search(params[:search])
+          active_rooms = BigBlueButtonApi.new.active_meetings
+          active_rooms_ids = []
+          participants = []
 
-          rooms.map! do |room|
-            {
-              friendly_id: room.friendly_id,
-              name: room.name,
-              owner: User.find(room.user_id).name
-            }
+          active_rooms.each do |active_room|
```

Comment:
I might create one `active_rooms_hash` data structure  that will store only the `meetingID => participantCount`.
The first advantage is that will have only one data structure that can do it all and the second advantage is that hashes are indexed data structures by nature meaning that finding an element doesn't require the traversal of all of its predecessors.
That means for the same number of active meetings a hash will always find the meeting in a constant time (in theory) while the array resource consumption will increase exponentially depending on the input size (This may not be crucial for our use case).
[More.](https://blog.sessionstack.com/how-javascript-works-arrays-vs-hash-tables-ab769bf84a2d#:~:text=Hash%20tables%20tend%20to%20be,the%20key%20and%20insert%20it.)
![image](https://user-images.githubusercontent.com/29759616/178471392-f4f603d0-825d-413b-a9b0-512a2bcca43e.png)